src/typescript-service.ts

```diff
@@ -1645,6 +1645,10 @@ export class TypeScriptService {
                     .filter((e): e is ts.DiagnosticWithLocation => !!e.file)
             )
         const diagnostics = iterate(tsDiagnostics)
+            // TS can report diagnostics without a file and range in some cases
+            // These cannot be represented as LSP Diagnostics since the range and URI is required
+            // https://github.com/Microsoft/TypeScript/issues/15666
+            .filter(diagnostic => !!diagnostic.file && diagnostic.start !== undefined && diagnostic.length !== undefined)
```

Comment:
So doesn't this filter out diagnostics without a range, whereas before they were reported as diagnostics with a range of 0:0-0:0?

Reply:
I see your point, so I moved the check. We still need to separate length == 0 from length == undefined in it.